lua/weapons/gmod_tool/stools/wire_adv.lua

```diff
@@ -413,6 +418,62 @@ elseif CLIENT then
 		return self.AutoWiringTypeLookup_t[inputtype]
 	end
 
+	function TOOL:IsWireEntity(ent)
+		-- TODO
+		local inputs, outputs = WireLib.GetPorts(ent)
+		return inputs or outputs
+	end
+
+	function TOOL:UpdateTraceForSurface(trace, parent, dir, terminate)
+		if self:GetClientNumber("stick") == 0 then return end
+		if not self:IsWireEntity(trace.Entity) then return end
+		terminate = terminate or false
+
+		dir = dir or 0
+		local traceData = util.GetPlayerTrace(LocalPlayer())
+		traceData.start = trace.HitPos
+		local normal
+		if dir == 1 then
+			normal = trace.Entity:GetForward()
+		elseif dir == 2 then
+			normal = trace.Entity:GetRight()
+		elseif dir == 3 then
+			normal = -trace.Entity:GetUp()
+		elseif dir == 4 then
+			normal = -trace.Entity:GetForward()
+		elseif dir == 5 then
+			normal = -trace.Entity:GetRight()
+		else -- if 0 then
+			normal = trace.Entity:GetUp()
+		end
```

Comment:
What is the point of the above 10-ish lines? Wouldn't this always be `normal = trace.HitNormal` at all times?

Reply:
No. This function traces the surface of the parent entity no matter where you hit the wire entity. For most gates the parent is in their "down" direction, but not for all. For some it's forward, etc. This function traces all directions from the hit point on the wire entity, to find the surface of the parent, to stick wires to their surface.

Reply:
Alright, but using recursion to solve this seems a bit strange. Personally I'd use regular old iteration instead to make it clearer what the code does, but the performance impact difference will be negligible.

Reply:
Alright.

Reply:
Decided to keep it. It'd be more messy as an iteration, since it has to restart, etc.